examples/resources/temporalcloud_metrics_endpoint/resource.tf

```diff
@@ -0,0 +1,69 @@
+terraform {
+  required_providers {
+    tls = {
+      source  = "hashicorp/tls"
+      version = ">= 2.0.0"
+    }
+    temporalcloud = {
+      source = "temporalio/temporalcloud"
+    }
+  }
+}
+
+provider "temporalcloud" {
+
+}
+
+// the following example demonstrates how to manage a metrics endpoint resource with a CA cert generated outside of Terraform
+
+resource "temporalcloud_metrics_endpoint" "terraform" {
+  accepted_client_ca = base64encode(file("${path.module}/ca.pem"))
+}
+
+// the following example demonstrates how to use the HashiCorp tls provider to generate certs for use in a metric endpoint
+// for more information see the provider's documentation here https://registry.terraform.io/providers/hashicorp/tls/latest/docs
+provider "tls" {
+}
+
+// root CA example - the metrics endpoint cert
+// This cert is not stored anywhere locally. 
+// If new certificates are needed you need to regenerate all of them (including the client end-entity certs).
+resource "tls_self_signed_cert" "ca" {
+  private_key_pem = tls_private_key.ca.private_key_pem
+  allowed_uses = [
+    "cert_signing",
+    "server_auth",
+    "client_auth",
+  ]
+  validity_period_hours = 8760 // 1 year
+  is_ca_certificate     = true
+}
+
+resource "tls_private_key" "default" {
+  algorithm = "RSA"
+}
+
+resource "tls_cert_request" "default" {
+  private_key_pem = tls_private_key.default.private_key_pem
+  dns_names       = []
+}
+
+// This is the end-entity cert that is used to authorize the workers connecting to temporal cloud.
+// Store this cert in KMS as a best practice
+// Reference your KMS's provider documentation for details on how to store a cert in KMS
+resource "tls_locally_signed_cert" "default" {
+  cert_request_pem      = tls_cert_request.default.cert_request_pem
+  ca_private_key_pem    = tls_private_key.ca.private_key_pem
+  ca_cert_pem           = tls_self_signed_cert.ca.cert_pem
+  validity_period_hours = var.certificate_expiration_hours
+  allowed_uses = [
+    "client_auth",
+    "digital_signature"
+  ]
+  is_ca_certificate = false
+}
+
+// example endpoint that uses the CA cert generated in this example
+resource "temporalcloud_metrics_endpoint" "terraform2" {
```

Comment:
What happens when you create 2 of the `temporalcloud_metrics_endpoint` resources? Which one will be applied to the account.

Is there a way to make this resource singleton? Not even sure if terraform supports such concept.

Reply:
@anekkanti there's no terraform support for singletons explicitly unfortunately, although we could put in some logic that does not allow you to create the resource if the endpoint is already configured and add some annotations stating this is an account singleton; this is what I've seen [recommended](https://discuss.hashicorp.com/t/how-to-write-a-terraform-provider-to-control-a-singleton-resource/34941/3) for similar situations.

Reply:
> put in some logic that does not allow you to create the resource if the endpoint is already configured and add some annotations stating this is an account singleton

I definitely like this, it will definitely elevate some of the confusion when someone accidentally creates multiple resources.